### PR TITLE
Enable CoreDNS as default - when k8s is installed with kube-up

### PR DESCRIPTION
**Pull request by fturib**
**DO NOT MERGE** 

**What this PR does / why we need it**:
Enable CoreDNS as default (for kube-up installations)
It will allow to run CI tests to prepare graduation criteria for CoreDNS as Default

see : KEP - https://github.com/kubernetes/community/blob/master/keps/sig-network/0012-20180518-coredns-default-proposal.md
see also : https://github.com/kubernetes/features/issues/566

NOTE for release : I guess that CoreDNS as default server for k8s needs a longer description. This specific PR is to ensure we validate all e2e.
If the 

```release-note
By default, when the cluster is setup using kube-up tool, 
the DNS Service of the cluster will be ensure by CoreDNS, instead of the usual kube-dns. 
However kube-dns can still be installed if the corresponding environment variable 
is set before launching the installation:
CLUSTER_DNS_CORE_DNS=false
```

### CODE REVIEW

**Comment by fturib** [comment]
@rajansandeep : finally most easy way is to do the change to DEFAULT, and run the /test.

**Comment by rajansandeep** [comment]
@MrHohn @bowei can this get an ok to test to be able to run the tests on the PR. Thanks!

**Comment by fturib** [comment]
Re-rerun the test now that it seems be ok more often (I mean basic issue fixed)
/test pull-kubernetes-e2e-gce-100-performance

**Comment by fturib** [comment]
Rerun the failing tests to check if that is coming from flakiness ... 
/test pull-kubernetes-e2e-kops-aws
/test pull-kubernetes-integration

**Comment by MrHohn** [comment]
Sorry for the delay, seems like the robot failed to request review...
/lgtm

What does the upgrade path from kube-dns to coredns look like? From GCE perspective, addon-manager will basically create coredns deployment, wait for a minute, and delete kube-dns deployment, which doesn't guarantee the number of available dns backends during the process. That is somehow concerning.

**Comment by fturib** [comment]
- The failing for "k8s-integration" is petty stable (although not all PRs) with an issues around the test : 
- The failing of "e2e-kops-aws" was not the same : first time was invalid one for deployment, second time : all tests themselved are OK, but the tear down of the cluster did not went ok : "k8s.io/kubernetes/test/integration/etcd TestEtcdStoragePath 41s"

**Comment by MrHohn** [comment]
@fturib I was referring to an upgrade of the cluster operated in the same way as what https://k8s-testgrid.appspot.com/sig-release-master-upgrade does, which is basically calling https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/upgrade.sh.

I honestly don't have knowledge about kubeadm upgrade, but would like to know what will happen there as well.

**Comment by MrHohn** [comment]
We are having more discussions on whether we should make this enabled by default at the same time as the graduation to GA. Holding it for now. @bowei @rramkumar1 @fturib @rajansandeep 
/hold

**Comment by bowei** [comment]
This should be held until the next release.

**Comment by fturib** [comment]
I will remove that PR, and re-push when needed for later release.

**Comment by fturib** [comment]
I Reopened this PR to validate that CoreDNS as "default" would pass the e2e tests.
We will keep it open until we get a GO from Sig-network to merge this PR.

File: cmd/kubeadm/app/phases/addons/dns/manifests.go [hunk 1]
```diff
@@ -244,8 +244,6 @@ spec:
         image: {{ .ImageRepository }}/coredns:{{ .Version }}
         imagePullPolicy: IfNotPresent
         resources:
-          limits:
```
Comment (johnbelamaric) [inline cmd/kubeadm/app/phases/addons/dns/manifests.go]: why are you removing limits?

**Comment by fturib** [comment]
@johnbelamaric : removing the limit was just a test ...

**Comment by fturib** [comment]
I close this PR in favor of this one : https://github.com/kubernetes/kubernetes/pull/67569

It has the same target, and fix issue https://github.com/kubernetes/dns/issues/254 found.

**Pull request closed without merging**